File: Learn/myrecyclerview/src/main/java/com/example/myrecyclerview/MainActivity.java
package com.example.myrecyclerview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import android.os.Bundle;
import android.util.Log;
import android.widget.LinearLayout;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private List<Bean> data = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        for (int i = 9000; i < 20000; i ++) {
            if (i % 4 != 0) {
                continue;
            }
            Bean bean = new Bean();
            bean.setName("可惜"+ i);
            data.add(bean);
        }

        RecyclerView recyclerView = findViewById(R.id.lv);

        //线性布局
//        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
//        recyclerView.setLayoutManager(linearLayoutManager);

        //网格布局
//        GridLayoutManager gridLayoutManager = new GridLayoutManager(this, 3);
//        recyclerView.setLayoutManager(gridLayoutManager);

        //瀑布流布局
        StaggeredGridLayoutManager staggeredGridLayoutManager = new StaggeredGridLayoutManager(3, LinearLayout.VERTICAL);
        recyclerView.setLayoutManager(staggeredGridLayoutManager);

        MyAdapter myAdapter = new MyAdapter(data, this);
        recyclerView.setAdapter(myAdapter);

        myAdapter.setOnRecyclerItemClickListener(new MyAdapter.OnRecyclerItemClickListener() {
            @Override
            public void onRecyclerItemClick(int position) {
                Log.d("itemClick", "position"+ position);
            }
        });
    }
}

File: Learn/recyclerview/src/main/java/com/example/recyclerview/MainActivity.kt
package com.example.recyclerview

import android.os.Bundle
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.StaggeredGridLayoutManager

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var data: MutableList<String> = mutableListOf<String>()
        for (i in 9000..19999) {
            if (i % 4 != 0) {
                continue
            }
            data.add("可惜$i")
        }
        var recyclerView = findViewById<RecyclerView>(R.id.rv)
        recyclerView.layoutManager = StaggeredGridLayoutManager(3, LinearLayout.VERTICAL)
        val adpater = MyAdapter(this, data)
        recyclerView.adapter = adpater
    }
}